### ISSUE TITLE
trtexec error: [pointWiseV2Builder.cpp::toExpr::325] Error Code 2: Internal Error (Assertion outScales.size() == 1 failed. )

### ISSUE THREAD

**Issue by HYLcool** [NVIDIA/TensorRT#2664]
Hi, I got 2 onnx models in QDQ format. There are only some differences on parameters "scale" in QuantizeLinear nodes. But when using following command to run them, one of them can be run successfully but the other can not.
```shell
trtexec --avgRuns=1000 --onnx=xxx.onnx --int8
```
The failed one prints errors like below:
```
......
[02/10/2023-17:59:09] [I] Finish parsing network model
[02/10/2023-17:59:09] [I] FP32 and INT8 precisions have been specified - more performance might be enabled by additionally specifying --fp16 or --best
[02/10/2023-17:59:09] [W] [TRT] Calibrator won't be used in explicit precision mode. Use quantization aware training to generate network with Quantize/Dequantize nodes.
[02/10/2023-17:59:11] [E] Error[2]: [pointWiseV2Builder.cpp::toExpr::325] Error Code 2: Internal Error (Assertion outScales.size() == 1 failed. )
[02/10/2023-17:59:11] [E] Error[2]: [builder.cpp::buildSerializedNetwork::636] Error Code 2: Internal Error (Assertion engine != nullptr failed. )
[02/10/2023-17:59:11] [E] Engine could not be created from network
[02/10/2023-17:59:12] [E] Building engine failed
[02/10/2023-17:59:12] [E] Failed to create engine from model or file.
[02/10/2023-17:59:12] [E] Engine set up failed
```
And the successful one prints normal logs like below:
```
......
[02/10/2023-17:56:53] [I] Finish parsing network model
[02/10/2023-17:56:53] [I] FP32 and INT8 precisions have been specified - more performance might be enabled by additionally specifying --fp16 or --best
[02/10/2023-17:56:53] [W] [TRT] Calibrator won't be used in explicit precision mode. Use quantization aware training to generate network with Quantize/Dequantize nodes.
[02/10/2023-17:56:57] [I] [TRT] [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +372, GPU +168, now: CPU 897, GPU 712 (MiB)
[02/10/2023-17:56:57] [I] [TRT] [MemUsageChange] Init cuDNN: CPU +119, GPU +54, now: CPU 1016, GPU 766 (MiB)
[02/10/2023-17:56:57] [I] [TRT] Local timing cache in use. Profiling results in this builder pass will not be stored.
^@^@^@[02/10/2023-17:58:31] [I] [TRT] Detected 1 inputs and 1 output network tensors.
[02/10/2023-17:58:31] [I] [TRT] Total Host Persistent Memory: 53328
[02/10/2023-17:58:31] [I] [TRT] Total Device Persistent Memory: 90624
[02/10/2023-17:58:31] [I] [TRT] Total Scratch Memory: 0
[02/10/2023-17:58:31] [I] [TRT] [MemUsageStats] Peak memory usage of TRT CPU/GPU memory allocators: CPU 57 MiB, GPU 12 MiB
[02/10/2023-17:58:31] [I] [TRT] [BlockAssignment] Algorithm ShiftNTopDown took 0.714037ms to assign 3 blocks to 34 nodes requiring 1204224 bytes.
[02/10/2023-17:58:31] [I] [TRT] Total Activation Memory: 1204224
[02/10/2023-17:58:31] [I] [TRT] [MemUsageChange] TensorRT-managed allocation in building engine: CPU +11, GPU +12, now: CPU 11, GPU 12 (MiB)
[02/10/2023-17:58:31] [W] [TRT] The getMaxBatchSize() function should not be used with an engine built from a network created with NetworkDefinitionCreationFlag::kEXPLICIT_BATCH flag. This function will always return 1.
[02/10/2023-17:58:31] [W] [TRT] The getMaxBatchSize() function should not be used with an engine built from a network created with NetworkDefinitionCreationFlag::kEXPLICIT_BATCH flag. This function will always return 1.
[02/10/2023-17:58:31] [I] Engine built in 100.458 sec.
[02/10/2023-17:58:31] [I] [TRT] [MemUsageChange] Init CUDA: CPU +0, GPU +0, now: CPU 814, GPU 708 (MiB)
[02/10/2023-17:58:31] [I] [TRT] Loaded engine size: 11 MiB
[02/10/2023-17:58:31] [I] [TRT] [MemUsageChange] TensorRT-managed allocation in engine deserialization: CPU +0, GPU +11, now: CPU 0, GPU 11 (MiB)
[02/10/2023-17:58:31] [I] Engine deserialized in 0.0110742 sec.
[02/10/2023-17:58:31] [I] [TRT] [MemUsageChange] TensorRT-managed allocation in IExecutionContext creation: CPU +0, GPU +1, now: CPU 0, GPU 12 (MiB)
[02/10/2023-17:58:31] [I] Using random values for input input.1
[02/10/2023-17:58:31] [I] Created input binding for input.1 with dimensions 1x3x224x224
[02/10/2023-17:58:31] [I] Using random values for output 191
[02/10/2023-17:58:31] [I] Created output binding for 191 with dimensions 1x1000
[02/10/2023-17:58:31] [I] Starting inference
......
```
Version information:
TensorRT: v8401
cuda: 11.7
cudnn: 8.5.0
cublas: 11.10.3.66

Could you help me to figure out that the failure reason? Thanks!

**Comment by zerollzeng** [comment]
`Error[2]: [pointWiseV2Builder.cpp::toExpr::325] Error Code 2: Internal Error (Assertion outScales.size() == 1 failed. )` is the error, can you share the problematic onnx here? Thanks!

**Comment by HYLcool** [comment]
Thanks for your reply! The problematic onnx models are available here:
- successful one: [link](https://drive.google.com/file/d/1iLxj5r-uevPy08-z1Cg1ZT8uGs87BuAO/view?usp=sharing)
- failed one: [link](https://drive.google.com/file/d/1huZEN5Ma-gTvNW2JSY7RdKlSO0lqHHCG/view?usp=sharing)

**Comment by HYLcool** [comment]
Hi @zerollzeng 

I find that this error is caused by the inconsistent quantization parameters between 133_scale and 132_scale as the figure below shows. I try to align their parameters (from 133 to 132 or vice versa) in the failed onnx model and it runs successfully.
<img width="523" alt="image" src="https://user-images.githubusercontent.com/12782861/219594545-4654de13-5a53-4bda-9417-53d0bcc562b9.png">


I think the failure might come from the branches of tensor 133 after the ReLU operator. After several experiments, I found that if there is no branch after ReLU, different quantization parameters before and after ReLU are allowed. However, if there are branches after ReLU, quantization parameters before and after ReLU should be the same. Otherwise, trtexec would print the errors mentioned above.

You can validate this thought on your own if you have any spare time. And could you please share an explanation from the TensorRT view? Thanks!

**Comment by zerollzeng** [comment]
Thanks for the info, I've created an internal bug to track this, id 3992985. Will come back to you if we have some progress.

**Comment by ttyio** [comment]
fixed in 8.6GA, closing and thanks!
